janus_server/src/aggregator.rs

```diff
@@ -8,68 +9,66 @@ use bytes::Bytes;
 use chrono::Duration;
 use http::{header::CACHE_CONTROL, StatusCode};
 use prio::codec::{Decode, Encode};
-use std::{
-    collections::HashMap,
-    convert::Infallible,
-    future::Future,
-    net::SocketAddr,
-    ops::Sub,
-    sync::{Arc, Mutex},
-};
+use std::{convert::Infallible, future::Future, net::SocketAddr, ops::Sub, sync::Arc};
 use tracing::warn;
 use warp::{filters::BoxedFilter, reply, trace, Filter, Rejection, Reply};
 
 /// Errors returned by functions and methods in this module
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
     /// An invalid configuration was passed.
-    #[error("Invalid configuration: {0}")]
+    #[error("invalid configuration: {0}")]
     InvalidConfiguration(&'static str),
     /// Error decoding an incoming message.
-    #[error("Message decoding failed: {0}")]
+    #[error("message decoding failed: {0}")]
     MessageDecode(#[from] prio::codec::CodecError),
     /// Corresponds to `staleReport`, §3.1
-    #[error("Stale report: {0}")]
+    #[error("stale report: {0}")]
     StaleReport(Nonce),
     /// Corresponds to `unrecognizedMessage`, §3.1
-    #[error("Unrecognized message: {0}")]
+    #[error("unrecognized message: {0}")]
     UnrecognizedMessage(&'static str),
     /// Corresponds to `outdatedHpkeConfig`, §3.1
-    #[error("Outdated HPKE config: {0}")]
+    #[error("outdated HPKE config: {0}")]
     OutdatedHpkeConfig(HpkeConfigId),
     /// A report was rejected becuase the timestamp is too far in the future,
     /// §4.3.4.
     // TODO(timg): define an error type in §3.1 and clarify language on
     // rejecting future reports
-    #[error("Report from the future: {0}")]
+    #[error("report from the future: {0}")]
     ReportFromTheFuture(Nonce),
+    #[error("datastore error: {0}")]
+    Datastore(#[from] datastore::Error),
```

Comment:
We shouldn't allow this conversion to happen automatically, as it would be easy to miss a call to `downcast_to_aggregator_error()`, and wind up with `crate::Error::Datastore(datastore::Error::User(crate::Error::...)`. In fact, we could turn `downcast_to_aggregator_error()` into our own `From` implementation.
```suggestion
    Datastore(datastore::Error),
```